### PR TITLE
TreeView: expand animation should work on expandAll (T1126510)

### PR DESCRIPTION
**Pull request by Zedwag**

### CODE REVIEW

File: js/ui/tree_view/ui.tree_view.base.js [hunk 1]
```diff
@@ -1626,10 +1626,23 @@ const TreeViewBase = HierarchicalCollectionWidget.inherit({
         }
     },
 
+    _allItemsExpanded: function() {
```
Comment (ksercs) [inline js/ui/tree_view/ui.tree_view.base.js]:
```suggestion
    _allItemsExpandedHandler: function() {
```

File: js/ui/tree_view/ui.tree_view.base.js [hunk 2]
```diff
@@ -1626,10 +1626,23 @@ const TreeViewBase = HierarchicalCollectionWidget.inherit({
...
-        each(dataAdapter.getData(), (_, node) => dataAdapter.toggleExpansion(node.internalFields.key, true));
-        this.repaint();
+        const nodes = this._dataAdapter.getData();
+        const expandingPromises = [];
+
+        this._skipContentReadyAndItemExpanded = true;
+
+        // NOTE: This is needed to support animation on expandAll, but stop triggering lot od contentReady/itemExpanded events.
+        // Deleting _allItemsExpanded() used to avoid error when component is disposed before promises fullfilles.
+
```
Comment (ksercs) [inline js/ui/tree_view/ui.tree_view.base.js]:
```suggestion
```

File: js/ui/tree_view/ui.tree_view.base.js [hunk 3]
```diff
@@ -1626,10 +1626,23 @@ const TreeViewBase = HierarchicalCollectionWidget.inherit({
...
     expandAll: function() {
-        const dataAdapter = this._dataAdapter;
-        each(dataAdapter.getData(), (_, node) => dataAdapter.toggleExpansion(node.internalFields.key, true));
-        this.repaint();
+        const nodes = this._dataAdapter.getData();
+        const expandingPromises = [];
+
+        this._skipContentReadyAndItemExpanded = true;
+
+        // NOTE: This is needed to support animation on expandAll, but stop triggering lot od contentReady/itemExpanded events.
```
Comment (ksercs) [inline js/ui/tree_view/ui.tree_view.base.js]: Please, review it with redactors

File: js/ui/tree_view/ui.tree_view.base.js [hunk 4]
```diff
@@ -1626,10 +1626,23 @@ const TreeViewBase = HierarchicalCollectionWidget.inherit({
...
+        // Deleting _allItemsExpanded() used to avoid error when component is disposed before promises fullfilles.
```
Comment (ksercs) [inline js/ui/tree_view/ui.tree_view.base.js]: I think it's a default practice and it shouldn't be described with a note
```suggestion
```

Comment (Zedwag) [reply js/ui/tree_view/ui.tree_view.base.js]: ok, deleted

File: js/ui/tree_view/ui.tree_view.base.js (hunk 3 above)
Comment (Zedwag) [reply js/ui/tree_view/ui.tree_view.base.js]: done, changed a bit

File: js/ui/tree_view/ui.tree_view.base.js (hunk 1 above)
Comment (Zedwag) [reply js/ui/tree_view/ui.tree_view.base.js]: done

File: js/ui/tree_view/ui.tree_view.base.js (hunk 2 above)
Comment (Zedwag) [reply js/ui/tree_view/ui.tree_view.base.js]: done

**Review by ksercs — APPROVED** [review]

**Review by EugeniyKiyashko — APPROVED** [review]

**Pull request merged**